File: 0189-rotate-array/0189-rotate-array.cs
public class Solution {
    public void Rotate(int[] nums, int k) {
        k = k % nums.Length;
        
        List<int> finalValues = new List<int>();
        
        for (int i = nums.Length - k; i < nums.Length; i++){
            finalValues.Add(nums[i]);
        }
        
        for (int i = nums.Length - k - 1; i >= 0; i--){
            nums[i + k] = nums[i];
        }
        
        for (int i = 0; i < finalValues.Count; i++){
            nums[i] = finalValues[i];
        }
    }
}

File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        if (k % nums.size() == 0) {return;}
        if (k > nums.size()){
            k = k % nums.size();
        }
        std::vector<int> subSet;
        int tmp;        
        for (int i = nums.size() - k; i < nums.size(); i++) {
            subSet.push_back(nums[i]);
        }
        for (int i = nums.size() - k - 1; i >= 0; i--) {
            nums[i + k] = nums[i];
        }
        for (int i = 0; i < k; i++) {
            nums[i] = subSet[i];
        }
    }
};
